scripts/run_ci_tests.sh

```diff
@@ -0,0 +1,262 @@
+EOF
+}
+
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    --mode)
+      MODE="${2:-}"
+      shift 2
+      ;;
+    --no-uv)
+      USE_UV="false"
+      shift 1
+      ;;
+    --skip-lint)
+      SKIP_LINT="true"
+      shift 1
+      ;;
+    --skip-unit)
+      SKIP_UNIT="true"
+      shift 1
+      ;;
+    --skip-functional)
+      SKIP_FUNCTIONAL="true"
+      shift 1
+      ;;
+    --gpus)
+      CUDA_VISIBLE_DEVICES="${2:-}"
+      shift 2
+      ;;
+    --hf-home)
+      HF_HOME="${2:-}"
+      shift 2
+      ;;
+    -h|--help)
+      usage
+      exit 0
+      ;;
+    *)
+      echo "Unknown option: $1" >&2
+      usage
+      exit 2
+      ;;
+  esac
+done
+
+export HF_HOME
+export CUDA_VISIBLE_DEVICES
+
+# Require GH_TOKEN to be set for operations that need GitHub API access.
+if [[ -z "${GH_TOKEN:-}" ]]; then
+  echo "[env] GH_TOKEN is not set. Please export GH_TOKEN before running this script." >&2
+  exit 1
+fi
+
+# Select tooling based on USE_UV or NO_UV
+if [[ "${USE_UV}" == "true" && -z "${NO_UV:-}" ]]; then
+  PYTHON="uv run python"
+  COVERAGE="uv run coverage"
+  PIP="uv pip"
+  PRECOMMIT="uv run pre-commit"
+  SYNC_CMD="uv sync --all-groups"
+else
+  PYTHON="python"
+  COVERAGE="python -m coverage"
+  PIP="pip"
+  PRECOMMIT="pre-commit"
+  SYNC_CMD="true"
+fi
+
+run_lint_local() {
+  if [[ "${SKIP_LINT}" == "true" ]]; then
+    echo "[lint] Skipped"
+    return 0
+  fi
+  ${PRECOMMIT} run --all-files --show-diff-on-failure --color=always
+}
+
+run_unit_local() {
+  if [[ "${SKIP_UNIT}" == "true" ]]; then
+    echo "[unit] Skipped"
+    return 0
+  fi
+  echo "[unit] Running unit tests with coverage"
+  ${COVERAGE} erase || true
+  ${COVERAGE} run -a -m pytest \
+    -o log_cli=true \
+    -o log_cli_level=INFO \
+    --disable-warnings \
+    -vs tests/unit_tests -m "not pleasefixme"
+}
+
+run_functional_local() {
+  if [[ "${SKIP_FUNCTIONAL}" == "true" ]]; then
+    echo "[functional] Skipped"
+    return 0
+  fi
+
+  echo "[functional] Training group (excluding inprocess restart)"
+  ${PYTHON} -m torch.distributed.run --nproc_per_node=2 --nnodes=1 -m coverage run -a -m pytest \
+    -o log_cli=true -o log_cli_level=INFO -v -s -x -m "not pleasefixme" --tb=short -rA \
+    tests/functional_tests/training -k "not test_inprocess_restart and not load_model"
+
+  if command -v ft_launcher >/dev/null 2>&1; then
+    echo "[functional] Inprocess restart with ft_launcher"
+    export TORCH_CPP_LOG_LEVEL="error"
+    ft_launcher \
+      --rdzv_backend=c10d --rdzv_endpoint=127.0.0.1:29500 \
+      --nnodes=1 --nproc-per-node=2 \
+      --ft-param-rank_section_timeouts=setup:600,step:180,checkpointing:420 \
+      --ft-param-rank_out_of_section_timeout=300 \
+      --monitor-interval=5 --max-restarts=3 \
+      --ft-restart-policy=min-healthy \
+      -m pytest -o log_cli=true -o log_cli_level=INFO -v -s -x -m "not pleasefixme" --tb=short -rA \
+      tests/functional_tests/training/test_inprocess_restart.py
```

Comment:
same here - as we need to sync changes from L2_training to this. maybe we can add an option to the script for wehther to use uv run python or not to reduce the duplication?

as a concrete example, we will need to add nsys profiling functional tests, which will require changes to L2_Launch_training that we'd then need to mirror here